libraries/Microsoft.Bot.Builder.Dialogs/DialogContext.cs

```diff
@@ -275,7 +275,7 @@ public async Task<DialogTurnResult> ContinueDialogAsync(CancellationToken cancel
         /// <seealso cref="Dialog.EndDialogAsync(ITurnContext, DialogInstance, DialogReason, CancellationToken)"/>
         public async Task<DialogTurnResult> EndDialogAsync(object result = null, CancellationToken cancellationToken = default(CancellationToken))
         {
-            if (result is CancellationToken token)
+            if (result is CancellationToken)
```

Comment:
This is a very strange line of code. It seems odd to call out CancellationToken as a special case. There are probably lots of things (wait handles, Tasks, delegates, an exception etc.) where it makes little or no sense, but such a categorization here seems bizarre.

Reply:
Having said that I'm fine with this PR - thanks for the clean up. I just merged this as is.